test/WebSites/InlineConstraintsWebSite/Startup.cs

```diff
@@ -12,6 +15,27 @@ public class Startup
         // Set up application services
         public void ConfigureServices(IServiceCollection services)
         {
+
+            services.Configure<RouteOptions>(
+                routeOptions => routeOptions.ConstraintMap.Add(
+                    "producttype",
+                    typeof(ItemTypeConstraint)));
+
+            services.Configure<RouteOptions>(
+                routeOptions => routeOptions.ConstraintMap.Add(
+                    "servicetype",
+                    typeof(ItemTypeConstraint)));
+
+            services.Configure<RouteOptions>(
+                routeOptions =>
+                {
+                    if (routeOptions.ConstraintMap.ContainsKey("servicetype"))
+                    {
+                        routeOptions.ConstraintMap["servicetype"] =
+                            typeof(ItemTypeUpperCaseConstraint);
+                    }
+                });
```

Comment:
Without the if statement, it will add the constraint to the constraint map. Since this sample or test is about updating an existing constraint, I checked if the key exists first.